src/nbl/video/IAPIConnection.cpp

```diff
@@ -43,7 +49,145 @@ IAPIConnection::IAPIConnection(const SFeatures& enabledFeatures)
         int ret = RENDERDOC_GetAPI(MinRenderdocVersion, (void**)&m_rdoc_api);
         assert(ret == 1);
     #endif
+
+        // probably is platform agnostic, for now
+        m_ngfx_api.injectNGFXToProcess();
+
+        m_debuggerType = m_ngfx_api.useNGFX ? EDT_NGFX :      // ngfx takes priority?
+            m_rdoc_api ? EDT_RENDERDOC : EDT_NONE;
```

Comment:
the functions to load the debugger should be moved out of the constructor into a method.

It should work more or less like this
```cpp
if (m_debuggerType!=EDebuggerType::None)
    LOG_ERROR("Graphics Debugger %d Already Attached!,m_debuggerType);
    
 #ifdef NBL_BUILD_WITH_NGFX
 #endif
 
 // if NGFX not loaded, then try Renderdoc
```